### DISCUSSION TITLE
Should rh-button always submit?

### DISCUSSION THREAD

**Discussion by bennypowers** [RedHat-UX/red-hat-design-system#1266] [Ideas]
## Consider

```html
<form>
  <button>Submit</button>
  <button>Show Logs</button>
</form>
```

Without custom javascript, the second button show above will submit the form

So too
```html
<form>
  <rh-button>Submit</rh-button>
  <rh-button variant="secondary">Show Logs</rh-button>
</form>
```
The secondary variant button here will submit the form

## Proposal
I submit (:drum:) that our design system buttons should request to submit the form if-and-only-if they have `type="submit"`. In all other cases, they should not submit the form without user-written custom javascript.

## Potential Pitfalls
submit behaviour of custom button FACEs has not yet been specified. If we take this decision now, we may come up against a future in which the standards have diverged. Nonetheless, in order to open up uses like "show logs" from above, I think we're within our rights to take this decision.

**Comment by kylebuch8** [comment]
I agree as long as we're explicit in our documentation that this differs from standard button functionality.

**Reply by bennypowers** [reply]
@nikkimk @hellogreg I'd value your feedback here. If both of you are in agreement with @kylebuch8 and myself, then I propose we promote this to an issue
